integration/shareddata/shareddata.go

```diff
@@ -197,6 +199,66 @@ func (values *Values[idType]) IsCompatible(backend string) bool {
 	return slices.Contains(values.backends, backend)
 }
 
+// fields is a map of field name -> value.
+type fields map[string]any
+
+// TopLevelValues stores shared data documents as {"_id": key, "field1": value1, "field2": value2, ...} documents.
+//
+//nolint:vet // for readability
+type TopLevelValues[id comparable] struct {
```

Comment:
I didn't come up with any good name for this provider, so suggestions are welcome!

Reply:
I would just call it something like "CustomeFields". Non-custom is for `v`, and custom is for any other case. But "CustomFields" doesn't look super perfect name, so I'm ok with your name too!